apm-agent-core/src/main/java/co/elastic/apm/agent/bci/InstrumentationStats.java

```diff
@@ -58,7 +62,9 @@ public Collection<String> getUsedInstrumentationGroups() {
         for (ElasticApmInstrumentation instrumentation : usedInstrumentations.keySet()) {
             usedInstrumentationGroups.addAll(instrumentation.getInstrumentationGroupNames());
         }
-        for (ElasticApmInstrumentation instrumentation : allInstrumentations) {
+        List<ElasticApmInstrumentation> allDeterministic = getAllInstrumentationsSorted();
+        for (ElasticApmInstrumentation instrumentation : allDeterministic) {
+            System.out.println("Instrumentation in order: " + instrumentation.getClass().getName());
```

Comment:
No more direct prints to std output please, only through [`SystemStandardOutputLogger`](https://github.com/elastic/apm-agent-java/blob/0e033282f2537c91e83e078c1506b1f763b5f956/apm-agent-common/src/main/java/co/elastic/apm/agent/common/util/SystemStandardOutputLogger.java#L29) please.
You may want to add a `DEBUG` level logging for this one.

Reply:
Yes, this is just for testing at the CI currently, I'll remove it when this PR is ready for review.
The test seems to be indeterminstic as in locally everything works and on the CI it doesn't, I'm trying to understand why